internal/servers/destinations.go

```diff
@@ -86,14 +86,14 @@ func (s *DestinationServer) Write2(msg pb.Destination_Write2Server) error {
 
 	for {
 		r, err := msg.Recv()
-		if err != nil {
-			if err == io.EOF {
-				close(resources)
-				if err := eg.Wait(); err != nil {
-					return fmt.Errorf("got EOF. failed to wait for plugin: %w", err)
-				}
-				return msg.SendAndClose(&pb.Write2_Response{})
+		if err == io.EOF {
+			close(resources)
```

Comment:
Let's move this outside the `for` loop and do a `defer` on it?
All exit routes do it, so I see no point in several places doing it.

Reply:
Maybe we can do each refactor in its own PR? That would make it easier to review and keep the changes in each PR small

Reply:
Ass this PR is about restructuring code for optimizing nesting, I think moving chan close outside would also allow to optimize the same code a bit better.

Reply:
Sure it's about refactoring, but if we can split the refactoring into multiple smaller PRs, is there a reason not to do it?